src/plugins/aws/idc/index.ts

```diff
@@ -0,0 +1,199 @@
+/** Copyright © 2024-present P0 Security
+
+This file is part of @p0security/cli
+
+@p0security/cli is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 3 of the License.
+
+@p0security/cli is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
+**/
+import { retryWithBackOff } from "../../../common/backoff";
+import { cached } from "../../../drivers/auth";
+import { print2 } from "../../../drivers/stdio";
+import {
+  AWSTokenResponse,
+  AWSClientInformation,
+  AWSAuthorizeResponse,
+} from "../../../types/aws/oidc";
+import { OidcLoginSteps } from "../../../types/oidc";
+import {
+  authorize,
+  DEVICE_GRANT_TYPE,
+  fetchOidcToken,
+  oidcLogin,
+  waitForActivation,
+} from "../../oidc/login";
+import { AwsCredentials } from "../types";
+import open from "open";
+
+export const registerClient = async (
+  region: string
```

Comment:
should we validate the region or is it always guaranteed to be valid?

Reply:
it should be valid, given this is the response from the idc call.